ctapipe/image/muon/muon_processor.py

```diff
@@ -0,0 +1,256 @@
+
+        self.source = EventSource(parent=self)
+        subarray = self.source.subarray
+
+        self.calib = CameraCalibrator(subarray=subarray, parent=self)
+        self.ring_fitter = MuonRingFitter(parent=self)
+        self.intensity_fitter = MuonIntensityFitter(subarray=subarray, parent=self)
+        self.cleaning = TailcutsImageCleaner(parent=self, subarray=subarray)
+
+        self.pixels_in_tel_frame = {}
+        self.field_of_view = {}
+        self.pixel_widths = {}
+
+        for p in ["min_pixels", "pedestal", "ratio_width", "completeness_threshold"]:
+            getattr(self, p).attach_subarray(self.source.subarray)
+
+    def __call__(self, event: ArrayEventContainer):
+        self._process_array_event(event)
+
+    def _process_array_event(self, event):
+        self.calib(event)
+        for tel_id, dl1 in event.dl1.tel.items():
+            self.process_telescope_event(event.index, tel_id, dl1)
+
+    def process_telescope_event(self, event_index, tel_id, dl1):
+        event_id = event_index.event_id
+
+        if self.source.subarray.tel[tel_id].optics.num_mirrors != 1:
+            self.log.warn(
+                f"Skipping non-single mirror telescope {tel_id}"
+                " set --allowed_tels to get rid of this warning"
+            )
+            return
+
+        self.log.debug(f"Processing event {event_id}, telescope {tel_id}")
+        image = dl1.image
+        if dl1.image_mask is None:
+            dl1.image_mask = self.cleaning(tel_id, image)
+
+        if np.count_nonzero(dl1.image_mask) <= self.min_pixels.tel[tel_id]:
+            self.log.debug(
+                f"Skipping event {event_id}-{tel_id}:"
+                f" has less then {self.min_pixels.tel[tel_id]} pixels after cleaning"
+            )
+            return
+
+        x, y = self.get_pixel_coords(tel_id)
+
+        # iterative ring fit.
+        # First use cleaning pixels, then only pixels close to the ring
+        # three iterations seems to be enough for most rings
+        mask = dl1.image_mask
+        for i in range(3):
+            ring = self.ring_fitter(x, y, image, mask)
+            dist = np.sqrt((x - ring.center_x) ** 2 + (y - ring.center_y) ** 2)
+            mask = np.abs(dist - ring.radius) / ring.radius < 0.4
+
+        if np.count_nonzero(mask) <= self.min_pixels.tel[tel_id]:
+            self.log.debug(
+                f"Skipping event {event_id}-{tel_id}:"
+                f" Less then {self.min_pixels.tel[tel_id]} pixels on ring"
+            )
+            return
+
+        if np.isnan(
+            [ring.radius.value, ring.center_x.value, ring.center_y.value]
+        ).any():
+            self.log.debug(
+                f"Skipping event {event_id}-{tel_id}: Ring fit did not succeed"
+            )
+            return
+
+        parameters = self.calculate_muon_parameters(tel_id, image, dl1.image_mask, ring)
+        # intensity_fitter does not support a mask yet, set ignored pixels to 0
```

Comment:
Can you fix this? This is already implemented. So just pass the mask to the fitter here.

Reply:
instead of zeroing the image

Reply:
Bump. Just remove the zeroing below and add `mask=mask` to the `intensitity_fitter` call below.

Reply:
This should allready have been implemented in f09c437 or did I have missed something here?